pkg/apis/sparkoperator.k8s.io/v1alpha1/types.go

```diff
@@ -200,6 +200,8 @@ type DriverSpec struct {
 	// Cores is used to set spark.driver.cores.
 	// Optional.
 	Cores *string `json:"cores,omitempty"`
+	// CoreLimit specifies a hard limit on CPU cores for the driver pod.
+	CoreLimit *string `json:"coreLimit,omitempty"`
```

Comment:
Wondering if it's just easier to take the podspec field from types.go and allow only a subset? Since that has all the possible parameterization possible.

Reply:
Yes, I was actually thinking about that. It seems a bit overkill as we really only allow setting a few fields. We can extend that through the initializer/mutation webhook, and I think we can revisit this later when we have better ideas on what fields should be customizable.